src/main/java/org/jabref/logic/autosaveandbackup/BackupManager.java

```diff
@@ -86,13 +87,27 @@ public static void shutdown(BibDatabaseContext bibDatabaseContext) {
     }
 
     /**
-     * Checks whether a backup file exists for the given database file.
+     * Checks whether a backup file exists for the given database file. If it exists, it is checked whether it is
+     * different from the original.
      *
-     * @param originalPath Path to the file a backup should be checked for.
+     * @param originalPath Path to the file a backup should be checked for. Example: jabref.bib.
+     * @return <code>true</code> if backup file exists AND differs from originalPath. <code>false</code> is the
+     * "default" return value in the good case. In the case of an exception <code>true</code> is returned to ensure that
+     * the user checks the output.
      */
-    public static boolean checkForBackupFile(Path originalPath) {
+    public static boolean backupFileDiffers(Path originalPath) {
         Path backupPath = getBackupPath(originalPath);
-        return Files.exists(backupPath) && !Files.isDirectory(backupPath);
+        if (!Files.exists(backupPath) || Files.isDirectory(backupPath)) {
+            return false;
+        }
+
+        try {
+            return !com.google.common.io.Files.equal(originalPath.toFile(), backupPath.toFile());
```

Comment:
Files.mismatch should do the job
https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/nio/file/Files.html#mismatch(java.nio.file.Path,java.nio.file.Path)